src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,99 @@
 
+public class StringManipulation implements StringManipulationInterface {
+    String string;
     @Override
     public String getString() {
         return null;
     }
 
     @Override
     public void setString(String string) {
+        this.string = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        StringTokenizer st = new StringTokenizer(string, " ");
+        return st.countTokens();
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        // if n < 0, throw illegal argument exception
+        if (n < 0) throw new IllegalArgumentException("n cannot be negative");
+        // if n > string.length(), throw out of bounds exception
+        else if (n > string.length()) throw new IndexOutOfBoundsException("n cannot exceed string length");
+        // if n == 0 the string is unchanged
+        else if (n == 0) return string;
+        // if every 1 char is being removed, all should be removed and either replaced with space or not
+        else if (n == 1) {
+            if (maintainSpacing) {
+                return " ".repeat(string.length());
+            }
+            else return "";
+        }
+        // otherwise, remove every nth character
+        else {
+            StringBuilder sb = new StringBuilder(string);
+            // start at end of string minus the remainder of the string divided by n, the "last" char to be removed
+            for (int i = string.length() - string.length() % n - 1; i >= 0; i -= n) {
+                if (maintainSpacing) {
+                    sb.replace(i,i+1," ");
+                }
+                else sb.deleteCharAt(i);
+            }
+            return sb.toString();
+        }
+
     }
 
     @Override
-    public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+    public String[] getSubStrings(int startWord, int endWord){
+        // if startword is after endword or either is less than or equal to 0, throw illegalargexcept
+        if (startWord > endWord || startWord <= 0)
+            throw new IllegalArgumentException("startword: " + startWord + " endword: " + endWord);
+        // convert string to stringtokenizer for easy counting and tokenizing
+        StringTokenizer st = new StringTokenizer(string, " ");
+        // if there are more words requested than present, throw indexoutofbounds
+        if (startWord > st.countTokens() || endWord > st.countTokens())
+            throw new IndexOutOfBoundsException("not enough words");
+        // convert tokenizer into String array
+        String[] substrings = new String[st.countTokens()];
+        int i = 0;
+        while (st.hasMoreTokens()) {
+            substrings[i++] = st.nextToken();
+        }
+        // return copy of array with only requested words
+        return Arrays.copyOfRange(substrings,startWord - 1,endWord);
     }
 
     @Override
-    public String restoreString(int[] indices) {
```

Comment:
You have two methods named restoreString .XD

Reply:
The - before the line indicates that it was removed from the last commit